memgpt/schemas/source.py

```diff
@@ -27,5 +27,28 @@ class Source(BaseSource):
     name: str = Field(..., description="The name of the source.")
 
 
+class SourceQuery(BaseSource):
+    id: Optional[str] = Field(None, description="The ID of the source.")
+    name: Optional[str] = Field(None, description="The name of the source.")
+
+
+class SourceIdQuery(BaseSource):
+    # TODO why is this necessary?
+    name: str = Field(..., description="The name of the source.")
+
+
 class SourceUpdate(BaseSource):
+    id: str = BaseSource.generate_id_field()  # TODO not sure if this is correct?
```

Comment:
@sarahwooders wasn't sure whether to use `= BaseSource.generate_id_field` or `: str = Field(...` for the `id` fields here

Reply:
Should be same as tools, so `BaseSource.generate_id_field`

Reply:
I saw `id` as a `str` here though - when to use `str` vs the other?

```py
 class ToolUpdate(ToolCreate):
    id: str = Field(..., description="The unique identifier of the tool.")
    name: Optional[str] = Field(None, description="The name of the function.")
    tags: Optional[List[str]] = Field(None, description="Metadata tags.")
    source_code: Optional[str] = Field(None, description="The source code of the function.")
    json_schema: Optional[Dict] = Field(None, description="The JSON schema of the function.")
```